variables.tf

```diff
@@ -77,82 +78,151 @@ variable "cluster_name" {
+  type        = any
+  default     = {}
+}
+#-------------------------------
+# EKS Cluster VPC Config
+#-------------------------------
+variable "cluster_endpoint_public_access" {
   type        = bool
-  default     = false
-  description = "Indicates whether or not the EKS private API server endpoint is enabled. Default to EKS resource and it is false"
+  default     = true
+  description = "Indicates whether or not the EKS public API server endpoint is enabled. Default to EKS resource and it is true"
 }
 
-variable "cluster_create_endpoint_private_access_sg_rule" {
+variable "cluster_endpoint_private_access" {
   type        = bool
   default     = false
-  description = "Whether to create security group rules for the access to the Amazon EKS private API server endpoint"
+  description = "Indicates whether or not the EKS private API server endpoint is enabled. Default to EKS resource and it is false"
 }
 
-variable "cluster_endpoint_private_access_cidrs" {
+variable "cluster_endpoint_public_access_cidrs" {
+  description = "List of CIDR blocks which can access the Amazon EKS public API server endpoint"
   type        = list(string)
-  default     = []
-  description = "List of CIDR blocks which can access the Amazon EKS private API server endpoint"
+  default     = ["0.0.0.0/0"]
 }
 
-variable "cluster_endpoint_private_access_sg" {
-  type        = list(string)
-  default     = []
-  description = "List of security group IDs which can access the Amazon EKS private API server endpoint"
+#-------------------------------
+# EKS Cluster ENCRYPTION
+#-------------------------------
+variable "cluster_kms_key_arn" {
+  type        = string
+  default     = null
+  description = "A valid EKS Cluster KMS Key ARN to encrypt Kubernetes secrets"
 }
 
-variable "cluster_endpoint_public_access" {
-  type        = bool
-  default     = true
-  description = "Indicates whether or not the EKS public API server endpoint is enabled. Default to EKS resource and it is true"
+variable "cluster_kms_key_deletion_window_in_days" {
+  type        = number
+  default     = 30
+  description = "The waiting period, specified in number of days (7 - 30). After the waiting period ends, AWS KMS deletes the KMS key"
 }
 
-variable "cluster_endpoint_public_access_cidrs" {
+variable "cluster_encryption_config" {
+  description = "Configuration block with encryption configuration for the cluster"
+  type = list(object({
+    provider_key_arn = string
+    resources        = list(string)
+  }))
+  default = []
+}
+#-------------------------------
+# EKS Cluster Kubernetes Network Config
+#-------------------------------
+variable "cluster_ip_family" {
+  description = "The IP family used to assign Kubernetes pod and service addresses. Valid values are `ipv4` (default) and `ipv6`. You can only specify an IP family when you create a cluster, changing this value will force a new cluster to be created"
+  type        = string
+  default     = null
```

Comment:
- default set to null instead of ipv4 as description says.
- can leverage `validation` to ensure it's one of the two strings (ipv4 / ipv6)

Reply:
I have used the core modules variables as is without any changes. We could try adding a validation. I will check this

Reply:
updated with validation